matchpy/matching/bipartite.py

```diff
@@ -154,17 +154,17 @@ def find_matching(self) -> Dict[TLeft, TRight]:
         # In addition, the graph stores which part of the bipartite graph a node originated from
         # to avoid problems when a value exists in both halfs.
         # Only one direction of the undirected edge is needed for the HopcroftKarp class
-        directed_graph = {}  # type: Dict[Tuple[int, TLeft], Set[Tuple[int, TRight]]]
+        directed_graph = {}  # type: Dict[Tuple[int, TLeft], List[Tuple[int, TRight]]]
 
         for (left, right) in self._edges:
             tail = (LEFT, left)
             head = (RIGHT, right)
             if tail not in directed_graph:
-                directed_graph[tail] = {head}
+                directed_graph[tail] = [head]
             else:
-                directed_graph[tail].add(head)
+                directed_graph[tail].append(head)
```

Comment:
It is not checked here if there are any duplicate elements. I would perhaps construct a second dictionary that maps to sets (like the old one) and use that to check if `head` was added to the list already. Or does the algorithm still work correctly if there are duplicates?

Reply:
There are no duplicates. `self._edges` is a dict, its keys are `(left, right)` which becomes the tuple `(tail, head)`. In this representation you can't have a duplicate `head` because keys of dictionaries are unique.